client/tcp_client.cpp

```diff
@@ -0,0 +1,63 @@
+#include "tcp_client.hpp"
+#include <future>
+#include <boost/asio/use_future.hpp>
+#include <array>
+#include <iostream>
+
+namespace fizyr
+{
+constexpr uint8_t BUFFER_SIZE = 128;
+
+tcp_client::tcp_client(const std::string& ip_addr, uint16_t port) :
+    ip_addr_(ip_addr), port_num_(port), work_(ios_), sock_(ios_)
+{
+    thread_ = std::make_unique<std::thread>([this](){ ios_.run(); });
```

Comment:
May be a comment here as to it is unique because it is associated with this client?

Reply:
Added comment.